OnboardingMarsV4/Pages/Language.cs

```diff
@@ -0,0 +1,246 @@
+﻿using MarsOnboardV2.Drivers;
+using MarsOnboardV2.Support;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.DevTools.V121.FedCm;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace MarsOnboardV2.Pages
+{
+    public class Language : CommonDriver
+    {
+        public IWebElement element;
+        public string xpathlasttable;
+        public string xpathduplicate1;
+        public string xpathduplicate2;
+        public string xpathupdate;
+
+        public Language()
+        {
+            element = null;
+            xpathlasttable = "//div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[1]";
+            xpathduplicate1 = "//div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[";
+            xpathduplicate2 = "]/tr/td[1]";
+            xpathupdate = "//div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[1]/tr/td[1]";
+
+        }
+
+        public IWebElement GetElement(string path)
+        {
+            element = driver.FindElement(By.XPath(path));
+            return element;
+        }
+        public string GetElementText(string path)
+        {
+            element = driver.FindElement(By.XPath(path));
+            return element.Text;
+        }
+
+        public void DeleteLanguage3()
+        {
+            Thread.Sleep(3000);
+
+            //checking if add button is present if not deleting the existing language
+            try
+            {
```

Comment:
This method seems weird to me can you please first deal with state I think I mentioned this before

State

example before adding language you need to make sure there a no existing languages as this could result in false positive test pass
you will need to think about this for all the scenarios

the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always
you can assume that test execution order is not always guaranteed
imagine if you run the tests  in parallel mode the all tests will run at the same time
so you need to also make sure you do not wipe out other tests data
hence why when you remove data the best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

Reply:
To consider state, in this version my strategy is:

1.To add a language, the test first checks for the availability of the "Add New" button. If the button is present, the language is added directly. If not, the test deletes the previously added language before clicking the "Add New" button and adding the new language. This dynamic check ensures that
a) There won't be any false positive (like English can be added but there is no add new button so it won't add and test will fail) 
b) It will avoid unnecessary actions of deleting all the previous data even added by other tester.

2. At the end of each test scenario, only the data added by the test itself is deleted to reset the system. This selective deletion prevents unintended interference with data added by other users or tests.

Same is strategy is applied with other Scenarios and they are running independently without deleting the other's test data, (except the last one that too if it is necessary).In Updating data I am checking availability of Add New button, adding language and updating it and at last deleting it.

In last version, I used to delete all existing language before adding/updating language then problem was if there is any data added by another user beforehand, all were deleted.